src/context/DamlHub.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { createContext } from 'react';
+
+import { usePolling } from '../utils';
+import { PartyToken } from '../party-info/PartyToken';
+import { fetchDefaultParties } from '../default-parties/DefaultParties';
+import { fetchPublicToken, PublicLedger } from '../public-streams/PublicLedger';
+
+interface DamlHubCtx {
+  publicToken?: PartyToken;
+  publicParty?: string;
+  userAdminParty?: string;
+}
+
+const initialDamlHubCtx = {
+  publicToken: undefined,
+  publicParty: undefined,
+  userAdminParty: undefined,
+};
+
+enum ActionType {
+  SET_PUBLIC_TOKEN = 'set-public-token',
+  SET_PUBLIC_PARTY = 'set-public-party',
+  SET_USERADMIN_PARTY = 'set-useradmin-party',
+}
+
+type Action =
+  | { type: ActionType.SET_PUBLIC_TOKEN; token: PartyToken }
+  | { type: ActionType.SET_PUBLIC_PARTY; party?: string }
+  | { type: ActionType.SET_USERADMIN_PARTY; party?: string };
+
+const reducer = (state: DamlHubCtx, action: Action): DamlHubCtx => {
+  switch (action.type) {
+    case ActionType.SET_PUBLIC_TOKEN:
+      return { ...state, publicToken: action.token };
+    case ActionType.SET_PUBLIC_PARTY:
+      return { ...state, publicParty: action.party };
+    case ActionType.SET_USERADMIN_PARTY:
+      return { ...state, userAdminParty: action.party };
+    default:
```

Comment:
Huh, actually it does! And in this case, it seems better to leave off `default`, because if I do forget to include a case, then it will actually yell at me during compilation

Reply:
Glad that I can be of use!